yapapi/executor/__init__.py

```diff
@@ -345,7 +347,7 @@ async def process_debit_notes() -> None:
                                 agr_id=debit_note.agreement_id, exc_info=sys.exc_info()  # type: ignore
                             )
                         )
-                if payment_closing and not agreements_to_pay:
+                if payment_closing and not debit_notes_accepted:
```

Comment:
I'd leave `agreements_to_pay` here. If there are no agreements to pay then we don't have to worry about debit notes. Note that `debit_notes_accepted` may be non-empty even if we don't wait for any more invoices, in this case this task would not stop until cancelled.